components/base/kinematicbase/differentialDrive.go

```diff
@@ -225,13 +228,25 @@ func (ddk *differentialDriveKinematics) issueCommand(ctx context.Context, curren
 	if err != nil {
 		return false, err
 	}
-	ddk.logger.Debug("distErr: %f\theadingErr %f", distErr, headingErr)
+	ddk.logger.Debugf("distErr: %.2f\theadingErr %.2f", distErr, headingErr)
 	if distErr > ddk.options.GoalRadiusMM && math.Abs(headingErr) > ddk.options.HeadingThresholdDegrees {
 		// base is headed off course; spin to correct
-		return true, ddk.Spin(ctx, math.Min(headingErr, ddk.options.MaxSpinAngleDeg), ddk.options.AngularVelocityDegsPerSec, nil)
+		err := ddk.Spin(ctx, math.Min(headingErr, ddk.options.MaxSpinAngleDeg), ddk.options.AngularVelocityDegsPerSec, nil)
+
+		if ddk.Localizer == nil {
```

Comment:
this seems a little dangerous. If there is an error reporting that we did not spin correctly, yet we tell the base to believe the spin was executed faithfully... This could potentially make `CheckPlan` not work..

Reply:
This a great point however, for explore mode this is fine. Since we have no localizer we will never be able to confirm if the spin action produces the correct orientation. 

The more important question is if this inconsistency would lead to issues with placing observed obstacles in the world frame. On this subject, we are fine as the incorrect orientation is applied to both the obstacle's position and the robot's position. An obstacle appearing in front of the robot in the incorrect orientation, would still have itself placed in front of the robot in the world frame represetantion.